runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -332,10 +341,16 @@ api.nvim_set_decoration_provider(namespace, {
             if type(label) == 'string' then
               text = label
             else
+              -- The label is made of diplay parts, so concatenate their values.
```

Comment:
```suggestion
              -- The label is made of diplay parts, so concatenate their values.
              --- @cast label lsp.InlayHintLabelPart[]
```

Reply:
In addition, I think exposing `lsp.InlayHintLabelPart[]` to API (rather than the "concatenated" labels) would be a more general solution. So the user-custom label formatter can directly access `label: string | InlayHintLabelPart[]` as specified in [the LSP Protocol](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#inlayHint).

Reply:
That's a good idea. We can also leave it to the user how they want to deal with display parts.

Reply:
I'm not fully typing label parts for now since I don't think we have the definitions for all subtypes yet.